### PR TITLE
install docker-ce-rootless-extras explicitly + unrecommend `sudo usermod -aG docker me`

### LINKED ISSUE #210: install.sh (rootful): update "If you would like to use Docker as a non-root user, you should now consider ..." text
https://github.com/docker/docker-install/blob/9dd5b52ae6ab9f316f59e828fbd75986df30b3e5/install.sh#L143-L155

This seems outdated. Guide for rootless mode should be appended or prepended here.

### LINKED ISSUE #212: deb: install docker-ce-rootless-extras automatically
Currently `docker-ce-rootelss-extras` is automatically installed on RPM distros but not installed on deb distros.

This is because [the deb spec does not specify `docker-ce-rootless-extras` as a hard dependency](https://github.com/docker/docker-ce-packaging/blob/4865d0386b312294a2a6f8dc5a0ed0bfbfff862c/deb/common/control#L37), while [the RPM spec specifies it as a hard dependency](https://github.com/docker/docker-ce-packaging/blob/4865d0386b312294a2a6f8dc5a0ed0bfbfff862c/rpm/SPECS/docker-ce.spec#L22). 

Thanks to @konstruktoid  for reporting this: https://github.com/docker/roadmap/issues/188

## Plan A
Remove `--no-install-recommends` https://github.com/docker/docker-install/blob/9dd5b52ae6ab9f316f59e828fbd75986df30b3e5/install.sh#L406

## Plan B
Add `docker-ce-rootless-extras$pkg_version` to this line:
https://github.com/docker/docker-install/blob/9dd5b52ae6ab9f316f59e828fbd75986df30b3e5/install.sh#L406

We may also want to add `slirp4netns` (>= 0.4.0) here because we do not have VPNkit in the deb package.
But `slirp4netns` (>= 0.4.0) deb is not available for Ubuntu 18.04 and Debian 10.
https://github.com/docker/docker-ce-packaging/blob/4865d0386b312294a2a6f8dc5a0ed0bfbfff862c/deb/common/control#L90
## Plan C
Move `docker-ce-rootless-extras` from `Recommends` to `Depends`: https://github.com/docker/docker-ce-packaging/blob/4865d0386b312294a2a6f8dc5a0ed0bfbfff862c/deb/common/control#L37

### PR DESCRIPTION
**Pull request by AkihiroSuda**
- install.sh: install docker-ce-rootless-extras explicitly

Fix #212

- install.sh: promote Rootless mode over "sudo usermod -aG docker me"

Fix #210

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@thaJeztah @tianon @tiborvass @tonistiigi PTAL?

File: install.sh [hunk 1]
```diff
@@ -140,8 +146,22 @@ echo_docker_as_nonroot() {
...
+	if [ -n "$has_rootless_extras" ]; then
+		echo
+		echo "===================================================================================================="
+		echo
+		echo "If you would like to use Docker as a non-root user, you should now consider"
+		echo "set up the Docker daemon (Rootless mode) for your user:"
+		echo
+		echo "  dockerd-rootless-setuptool.sh install"
+		echo
+		echo "For more information about Rootless mode, refer to https://docs.docker.com/engine/security/rootless/"
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply install.sh:158]: Have some other suggestions, but let me open a PR against your branch, that's probably easier 😅

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by thaJeztah** [comment]
Opened https://github.com/AkihiroSuda/docker-install/pull/2 against your fork with suggestions 🤗

**Comment by AkihiroSuda** [comment]
Thanks, merged https://github.com/AkihiroSuda/docker-install/pull/2

**Review by thaJeztah — APPROVED** [review]
LGTM

@tiborvass @StefanScherer ptal

**Review by StefanScherer — APPROVED** [review]
LGTM ❤️

**Comment by thaJeztah** [comment]
We can merge this after https://github.com/docker/docker.github.io/pull/12387 is merged (which makes the new redirects go live on docs.docker.com)

**Comment by thaJeztah** [comment]
Tested the script on Ubuntu 20.10, and install was successfully, so going ahead and merge this one

**Pull request merged**